### ISSUE TITLE
Exact State of Environment in Demonstrations

### ISSUE THREAD

**Issue by arjung128** [haosulab/ManiSkill-Learn#15]
Do we know the exact state of the environment in these demonstrations (i.e. position / orientation of articulated object)? Can this environment state scene be loaded into the visualizer? Or do we simply know the level of the environment used in the demonstrations (https://github.com/haosulab/ManiSkill-Learn/issues/8)?

**Comment by xuanlinli17** [comment]
Yes, the exact state of the environment is provided in `state` mode of the demonstrations. Please refer to https://github.com/haosulab/ManiSkill-Learn#generating-custom-point-cloud-demonstrations for more info. We can also visualize the environments by setting the env state using `env.set_state` (similar to https://github.com/haosulab/ManiSkill-Learn/blob/main/tools/convert_state.py , and visualization can be seen when `save_video` in https://github.com/haosulab/ManiSkill-Learn/blob/main/mani_skill_learn/env/evaluation.py)

**Comment by arjung128** [comment]
Thanks for your prompt response!

I am able to run the evaluation script to see the performance of the pre-trained model on OpenCabinetDrawer. The environment name is `OpenCabinetDrawer_1045_link_0-v0` and the level is `2153004774`. Additionally, `self.env.get_state()` returns a 94-dim vector. Can all of this information be used to extract the exact articulated object asset_id and pose, such that I can use the following lines of code to replicate it? What about the camera pose?

```
sapien_assets_id = 41083
urdf = sapien_og.asset.download_partnet_mobility(sapien_assets_id, token=my_token)
asset = loader.load_kinematic(urdf)
asset.set_pose(sapien.Pose([1.0155, 0.3, -0.1]))
```

I know the exact robot pose can be extracted from `obs['state']`, but how exactly can this be done? `obs['state']` is a 38-dim vector -- what does each element correspond to? `self.robot.set_root_pose()`,` self.robot.set_qpos()`, `self.active_joints[j].set_drive_target()` and `self.active_joints[j].set_drive_velocity_target()` were used in the motion planning tutorial to set the robot pose -- can these same functions be used to set the robot to have the same pose as the one given by `obs['state']`?

I am asking these things because I want to replicate this environment by manually adding the robot, the cabinet, etc. to the scene (using `self.robot.set_root_pose()`, `asset.set_pose()`, etc., in the same way all of these things were added to the scene in the motion planning tutorial) such that it can be visualized using the interactive Viewer. Once this is done, then I intend to try to use motion planning to solve the task for a direct comparison.

**Comment by lz1oceani** [comment]
env.get_state will return the internal state of the environment which is for reset the environment. It does not cover the object id. The object id is specified by env.reset(level). Our camera is attached on the robot, directly set the pose may move the robot too.
